### PR TITLE
grouped method

### PR DESCRIPTION
**Pull request by luca-vercelli**
This PR will add a new method grouped() to all models. This method is useful, IMHO, for grouping data in reports. It is a high-level method, as opposed to the low-level method read_group().

For example, in future it will be possible to simplify the code that currently is used to populate the 
two little tables under invoice reports.

No impact at all in previous existing code.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/models.py [hunk 1]
```diff
@@ -5005,6 +5005,170 @@ def sorted(self, key=None, reverse=False):
...
+                groups.append(group)
+
+            for group in groups:
+                for fname, f in aggregate_functions.iteritems():
+                    group.__dict__[fname] = f(group._all_records)
+
+            for group in groups:
+                group.items = group._all_records._grouped(groupby[1:], orderby[1:], **aggregate_functions)
+
+            return sorted(groups, key=orderby[0])
```
Comment (mreficent) [inline odoo/models.py]: put a blank line below

**Comment by luca-vercelli** [comment]
I still have an error from pylint. Can somebody tell me why eval() is forbidden? that's a great limitation.

**Comment by mreficent** [comment]
ping: @xmo-odoo

**Comment by mart-e** [comment]
> I still have an error from pylint. Can somebody tell me why eval() is forbidden? that's a great limitation.

For security reason. Also, this should definitely target the master branch.

cc @rco-odoo

**Comment by luca-vercelli** [comment]
@mart-e Thank you. Now all checks have passed.
w.r.t. branch "master", my problem is that I have no working 11.0 installation here. I am not sure I can migrate this code as-is from 10.0 to 11.0.

**Pull request closed without merging**